### ISSUE TITLE
Unable to initialize windows kex seamless mode

### ISSUE THREAD

**Issue by Cr7pt0nic** [microsoft/WSL#9333]
### Version

Windows NT 10.0 build 22000 (64-bit)

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.79.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Wed Nov 23 01:01:46 UTC 2022

### Distro Version

Kali Linux 2022.4

### Other Software

For some weird reason whenever running Kali Linux Kex on seamless mode I get this error

![image](https://user-images.githubusercontent.com/48302509/207749902-52472617-e2d3-4a95-b64f-d3608cf85e3f.png)

Here's the log information:

`Welcome to the VcXsrv X Server
Vendor: The VcXsrv Project
Release: 1.20.14.0

OS: Windows NT 10.0 build 22000 (64-bit)
Contact: marha@users.sourceforge.net

LoadPreferences: C:\Users\57pat\AppData\Roaming\.XWinrc not found
LoadPreferences: Loading \\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\system.XWinrc
(II) AIGLX: Testing pixelFormatIndex 5
(II) GLX: enabled GLX_SGI_make_current_read
(II) GLX: enabled GLX_SGI_swap_control
(II) GLX: enabled GLX_MESA_swap_control
(II) GLX: enabled GLX_SGIX_pbuffer
(II) GLX: enabled GLX_ARB_multisample
(II) GLX: enabled GLX_SGIS_multisample
(II) GLX: enabled GLX_ARB_fbconfig_float
(II) GLX: enabled GLX_EXT_fbconfig_packed_float
(II) GLX: enabled GLX_ARB_create_context
(II) GLX: enabled GLX_ARB_create_context_profile
(II) GLX: enabled GLX_ARB_create_context_robustness
(II) GLX: enabled GLX_EXT_create_context_es2_profile
(II) AIGLX: enabled GLX_MESA_copy_sub_buffer
(II) 670 pixel formats reported by wglGetPixelFormatAttribivARB
(II) GLX: Initialized Win32 native WGL GL provider for screen 0
[xkb] Starting '"\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbcomp" -w 1 "-R\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbdata" -xkm "C:\Users\57pat\AppData\Local\Temp\xkb_a06412" -em1 "The XKEYBOARD keymap compiler (xkbcomp) reports:" -emp "> " -eml "Errors from xkbcomp are not fatal to the X server" "C:\Users\57pat\AppData\Local\Temp\server-3.xkm"' failed: Incorrect function.
(EE) Error compiling keymap (server-3) executing '"\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbcomp" -w 1 "-R\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbdata" -xkm "C:\Users\57pat\AppData\Local\Temp\xkb_a06412" -em1 "The XKEYBOARD keymap compiler (xkbcomp) reports:" -emp "> " -eml "Errors from xkbcomp are not fatal to the X server" "C:\Users\57pat\AppData\Local\Temp\server-3.xkm"'
(EE) XKB: Couldn't compile keymap
(EE) XKB: Failed to load keymap. Loading default keymap instead.
[xkb] Starting '"\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbcomp" -w 1 "-R\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbdata" -xkm "C:\Users\57pat\AppData\Local\Temp\xkb_a06412" -em1 "The XKEYBOARD keymap compiler (xkbcomp) reports:" -emp "> " -eml "Errors from xkbcomp are not fatal to the X server" "C:\Users\57pat\AppData\Local\Temp\server-3.xkm"' failed: Incorrect function.
(EE) Error compiling keymap (server-3) executing '"\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbcomp" -w 1 "-R\\wsl.localhost\kali-linux\usr\lib\win-kex\VcXsrv\xkbdata" -xkm "C:\Users\57pat\AppData\Local\Temp\xkb_a06412" -em1 "The XKEYBOARD keymap compiler (xkbcomp) reports:" -emp "> " -eml "Errors from xkbcomp are not fatal to the X server" "C:\Users\57pat\AppData\Local\Temp\server-3.xkm"'
(EE) XKB: Couldn't compile keymap
XKB: Failed to compile keymap
Keyboard initialization failed. This could be a missing or incorrect setup of xkeyboard-config.
(EE)
Fatal server error:
(EE) Failed to activate virtual core keyboard: 2(EE)
(EE) Server terminated with error (1). Closing log file.`

### Repro Steps

I ran

kex --sl <-- which gave me the same error

sudo kex --sl <-- still same error

sudo kex --sl --wtstart -s <--- which also gave me the same error

### Expected Behavior

It's expected behavior is to popup a linux GUI on the top bar

![image](https://user-images.githubusercontent.com/48302509/207750301-f9a3fdfc-9927-4dc9-9beb-1d2b49be99d4.png)

### Actual Behavior

The actual behavior that happens is this instead:

![image](https://user-images.githubusercontent.com/48302509/207750354-e75e6ed7-9cf1-4753-9da8-6d0cc2b814f0.png)

**Comment by rescenic** [comment]
Yeah it's well known problem. Just use full screen mode or Enhanced Session Mode. Klik Logout to exit full screen mode.

**Comment by Cr7pt0nic** [comment]
I've actually tried everything so far and I even tried using Fullscreen mode or enhanced session and it still doesn't work. It's a problem with the application itself and it needs to be fixed. This is my main system that I use on an everyday basis and I hate having to switch between virtual machines every single time and it's absolutely frustrating.

**Comment by Cr7pt0nic** [comment]
Literally all I did was update WSL and for some reason kex just completely breaks on itself

**Comment by dethknite** [comment]
Same. "kex --sl --wtstart -s" was working fine up until about 3 weeks ago (after windows updates).

**Comment by theonlyfoxy** [comment]
Same issue here, some guy suggested to not use Store's WSL version.

**Comment by dethknite** [comment]
Yeah.. there are some open issues on the store version and the mount of the windows binaries. I imagine (or hope) those are being fixed.. though uninstalling it and installing the builtin does solve it for now.

# Cmd to install windows builtin wsl after removing store version
wsl --install --inbox

**Comment by hanshou101** [comment]
> Yeah.. there are some open issues on the store version and the mount of the windows binaries. I imagine (or hope) those are being fixed.. though uninstalling it and installing the builtin does solve it for now.
> 
> Cmd to install windows builtin wsl after removing store version `wsl --install --inbox`

Thank you so much. I uninstalled 【WSL】 and 【Kali-Linux】 in the 【Control Panel】, and after 【wsl.exe --unregister kali-linux】, then follow your method to perform 【wsl --install --inbox】 . After reinstalling the 【kali-linux】 subsystem, all problems have been resolved.

**Comment by GiggleMonster** [comment]
> > Yeah.. there are some open issues on the store version and the mount of the windows binaries. I imagine (or hope) those are being fixed.. though uninstalling it and installing the builtin does solve it for now.
> > Cmd to install windows builtin wsl after removing store version `wsl --install --inbox`
> 
> Thank you so much. I uninstalled 【WSL】 and 【Kali-Linux】 in the 【Control Panel】, and after 【wsl.exe --unregister kali-linux】, then follow your method to perform 【wsl --install --inbox】 . After 【wsl --install -d kali-linux】 reinstalling the subsystem, all problems have been resolved.

Please keep in mind that unregistering will delete all of your data.

**Comment by ITProggeruz** [comment]
now im pain about to: i installed that stupid kex in a hour and other things for 3 hours
